ipa-core/src/config.rs

```diff
@@ -631,4 +782,308 @@ mod tests {
+cT/hDCXCoob4pJ/fpPDMrqhiwTTck3bNOuzv9QIx+p5C2Qp8u67rYfK78w86NaNZ
+MFcwJgYDVR0RBB8wHYIbaGVscGVyMS5wcm9kLmlwYS1oZWxwZXIuZGV2MA4GA1Ud
+DwEB/wQEAwICpDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwCgYIKoZI
+zj0EAwIDSAAwRQIhAKVdDCQeXLRXDYXy4b1N1UxD/JPuD9H7zeRb8/nmIDTfAiBL
+a6L0t1Ug8i2RcequSo21x319Tvs5nUbGwzMFSS5wKA==
+-----END CERTIFICATE-----
+\"\"\"
+url = \"helper1.prod.ipa-helper.dev:443\"";
+
+    /// The rest of a configuration
+    const REST: &str = "
+[peers.hpke]
+public_key = \"f458d5e1989b2b8f5dacd4143276aa81eaacf7449744ab1251ff667c43550756\"
+
+[[peers]]
+certificate = \"\"\"
+-----BEGIN CERTIFICATE-----
+MIIBmzCCAUGgAwIBAgIITOtoca16QckwCgYIKoZIzj0EAwIwJjEkMCIGA1UEAwwb
+aGVscGVyMi5wcm9kLmlwYS1oZWxwZXIuZGV2MB4XDTI0MDkwNDAzMzMwOFoXDTI0
+MTIwNDAzMzMwOFowJjEkMCIGA1UEAwwbaGVscGVyMi5wcm9kLmlwYS1oZWxwZXIu
+ZGV2MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAETxOH4ATz6kBxLuRznKDFRugm
+XKmH7mzRB9wn5vaVlVpDzf4nDHJ+TTzSS6Lb3YLsA7jrXDx+W7xPLGow1+9FNqNZ
+MFcwJgYDVR0RBB8wHYIbaGVscGVyMi5wcm9kLmlwYS1oZWxwZXIuZGV2MA4GA1Ud
+DwEB/wQEAwICpDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwCgYIKoZI
+zj0EAwIDSAAwRQIhAI4G5ICVm+v5KK5Y8WVetThtNCXGykUBAM1eE973FBOUAiAS
+XXgJe9q9hAfHf0puZbv0j0tGY3BiqCkJJaLvK7ba+g==
+-----END CERTIFICATE-----
+\"\"\"
+url = \"helper2.prod.ipa-helper.dev:443\"
+
+[peers.hpke]
+public_key = \"62357179868e5594372b801ddf282c8523806a868a2bff2685f66aa05ffd6c22\"
+
+[[peers]]
+certificate = \"\"\"
+-----BEGIN CERTIFICATE-----
+MIIBmzCCAUGgAwIBAgIIaf7eDCnXh2swCgYIKoZIzj0EAwIwJjEkMCIGA1UEAwwb
+aGVscGVyMy5wcm9kLmlwYS1oZWxwZXIuZGV2MB4XDTI0MDkwNDAzMzMxMloXDTI0
+MTIwNDAzMzMxMlowJjEkMCIGA1UEAwwbaGVscGVyMy5wcm9kLmlwYS1oZWxwZXIu
+ZGV2MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEIMqxCCtu4joFr8YtOrEtq230
+NuTtUAaJHIHNtv4CvpUcbtlFMWFYUUum7d22A8YTfUeccG5PsjjCoQG/dhhSbKNZ
+MFcwJgYDVR0RBB8wHYIbaGVscGVyMy5wcm9kLmlwYS1oZWxwZXIuZGV2MA4GA1Ud
+DwEB/wQEAwICpDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwCgYIKoZI
+zj0EAwIDSAAwRQIhAOTSQWbN7kfIatNJEwWTBL4xOY88E3+SOnBNExCsTkQuAiBB
+/cwOQQUEeE4llrDp+EnyGbzmVm5bINz8gePIxkKqog==
+-----END CERTIFICATE-----
+\"\"\"
+url = \"helper3.prod.ipa-helper.dev:443\"
+
+[peers.hpke]
+public_key = \"55f87a8794b4de9a60f8ede9ed000f5f10c028e22390922efc4fb63bc6be0a61\"
+";
+
+    /// Valid: A sharded configuration
+    const SHARDED_OK: &str = "
+[[peers]]
+certificate = \"\"\"
+-----BEGIN CERTIFICATE-----
+MIIBmzCCAUGgAwIBAgIIMlnveFys5QUwCgYIKoZIzj0EAwIwJjEkMCIGA1UEAwwb
+aGVscGVyMS5wcm9kLmlwYS1oZWxwZXIuZGV2MB4XDTI0MDkwNDAzMzMwM1oXDTI0
+MTIwNDAzMzMwM1owJjEkMCIGA1UEAwwbaGVscGVyMS5wcm9kLmlwYS1oZWxwZXIu
+ZGV2MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEWmrrkaKM7HQ0Y3ZGJtHB7vfG
+cT/hDCXCoob4pJ/fpPDMrqhiwTTck3bNOuzv9QIx+p5C2Qp8u67rYfK78w86NaNZ
+MFcwJgYDVR0RBB8wHYIbaGVscGVyMS5wcm9kLmlwYS1oZWxwZXIuZGV2MA4GA1Ud
+DwEB/wQEAwICpDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwCgYIKoZI
+zj0EAwIDSAAwRQIhAKVdDCQeXLRXDYXy4b1N1UxD/JPuD9H7zeRb8/nmIDTfAiBL
+a6L0t1Ug8i2RcequSo21x319Tvs5nUbGwzMFSS5wKA==
+-----END CERTIFICATE-----
+\"\"\"
+url = \"helper1.prod.ipa-helper.shard0.dev:443\"
+shard_port = 444
```

Comment:
The inconsistency between ways to specify shard and mpc ports is worrying but not enough to block this change.

Reply:
I'll use sharded_url then